raiden/transfer/merkle_tree.py

```diff
@@ -22,11 +22,12 @@ def hash_pair(first: Optional[Keccak256], second: Optional[Keccak256]) -> Option
     deterministic way of ordering the elements making sure the smart contract
     verification and the python code are compatible.
     """
-
     if first is None:
+        assert second, 'Both arguments to hash_pair() were None'
         return second
 
     if second is None:
+        assert first, 'Both arguments to hash_pair() were None'
```

Comment:
This looks like the wrong approach. Why are any of the arguments optional anyways?

Reply:
To be honest that was exactly my question too. But I opened the PR to get feedback on this. If you also think these can't be optional then I guess they are not. Checking the code, I don't see how with the current codebase they are optional. They are always given if I read it correctly.

When the function was introduced [here](https://github.com/raiden-network/raiden/commit/ea3a6c4508feeba8d52e689c044a91777665094d#diff-5a4eee2a09172ec6d82c5be1929443a4R21) it was introduced without types, but with the argument being checked against `None` so I guess the types followed that without much thinkng.